index.js

```diff
@@ -1,48 +1,53 @@
 'use strict'
 const json = require('fast-json-stable-stringify')
-const CID = require('cids')
 const isCircular = require('is-circular')
 const transform = require('lodash.transform')
-const codecInterface = require('@ipld/codec-interface')
 
-const _encode = obj => transform(obj, (result, value, key) => {
-  if (CID.isCID(value)) {
-    result[key] = { '/': value.toBaseEncodedString() }
-  } else if (Buffer.isBuffer(value)) {
-    result[key] = { '/': { base64: value.toString('base64') } }
-  } else if (typeof value === 'object' && value !== null) {
-    result[key] = _encode(value)
-  } else {
-    result[key] = value
-  }
-})
+module.exports = multiformats => {
+  const { CID, bytes, multibase } = multiformats
+  if (!multibase.has('base64')) throw new Error('Multibase must include base64')
+  if (!multibase.has('base32')) throw new Error('Multibase must include base32')
+  const _encode = obj => transform(obj, (result, value, key) => {
+    if (CID.isCID(value)) {
+      result[key] = { '/': value.toString() }
+    } else if (bytes.isBinary(value)) {
+      value = bytes.coerce(value)
+      result[key] = { '/': { base64: multibase.encode(value, 'base64') } }
+    } else if (typeof value === 'object' && value !== null) {
+      result[key] = _encode(value)
+    } else {
+      result[key] = value
+    }
+  })
 
-const encode = obj => {
-  if (isCircular(obj)) {
-    throw new Error('Object contains circular references.')
-  }
+  const encode = obj => {
+    if (isCircular(obj)) {
+      throw new Error('Object contains circular references.')
+    }
 
-  const data = _encode(obj)
-  return Buffer.from(json(data))
-}
+    const data = _encode(obj)
+    return bytes.fromString(json(data))
+  }
 
-const _decode = obj => transform(obj, (result, value, key) => {
-  if (typeof value === 'object' && value !== null) {
-    if (value['/']) {
-      if (typeof value['/'] === 'string') result[key] = new CID(value['/'])
-      else if (typeof value['/'] === 'object' && value['/'].base64) {
-        result[key] = Buffer.from(value['/'].base64, 'base64')
-      } else result[key] = _decode(value)
+  const _decode = obj => transform(obj, (result, value, key) => {
```

Comment:
I bring it up as we talked before, that I find your code hard to read and I want to show you when and why I feel like it. I would write `_decode` like this:

```js
const _decode = obj => transform(obj, (result, value, key) => {
  if (typeof value === 'object' && value !== null) {
    if (value['/']) {
      if (typeof value['/'] === 'string') {
        result[key] = new CID(value['/'])
      } else if (typeof value['/'] === 'object' && value['/'].base64) {
        result[key] = multibase.decode(value['/'].base64, 'base64')
      } else {
        result[key] = _decode(value)
      }
    } else {
      result[key] = _decode(value)
    }
  } else {
    result[key] = value
  }
})
```

Compared to your version, I can quickly see with a vertical scan skipping over the actual if/else statements, that all arms of these nested if/else change the `result[key]`. No arm is doing something else. If I want to know more, I look deeper on the if statements. I can also quickly see that the default case (the final `else` arms) is to recursively call `_decode()` if it's an object, if it's something else, I just use that value.

And for even quicker reading with comments:

```js
const _decode = obj => transform(obj, (result, value, key) => {
  if (typeof value === 'object' && value !== null) {
    if (value['/']) {
      // It's a CID
      if (typeof value['/'] === 'string') {
        result[key] = new CID(value['/'])
      }
      // It's bytes
      else if (typeof value['/'] === 'object' && value['/'].base64) {
        result[key] = multibase.decode(value['/'].base64, 'base64')
      } else {
        result[key] = _decode(value)
      }
    } else {
      result[key] = _decode(value)
    }
  } else {
    result[key] = value
  }
})
```

Reply:
Ya, I can’t actually argue that the single line `if` statements here make it **more** readable, so I changed it.